### PR TITLE
Expose `MergeResult` from #4273 via `VersionStore`

### PR DESCRIPTION
**Pull request by snazy**

### CODE REVIEW

File: model/src/test/java/org/projectnessie/model/TestModelObjectsSerialization.java [hunk 1]
```diff
@@ -47,7 +49,15 @@ public class TestModelObjectsSerialization {
   @MethodSource("goodCases")
   void testGoodSerDeCases(Case goodCase) throws IOException {
     String json = MAPPER.writeValueAsString(goodCase.obj);
-    Assertions.assertThat(json).isEqualTo(goodCase.deserializedJson);
+    JsonNode j;
```
Comment (snazy) [inline model/src/test/java/org/projectnessie/model/TestModelObjectsSerialization.java:52]: String comparison is flaky/error-prone, because the actual order of elements might change.
Parsing the strings as `JsonNode`s is safe though.

**Review by dimas-b — APPROVED** [review]

File: model/src/test/java/org/projectnessie/model/TestModelObjectsSerialization.java [hunk 2]
```diff
@@ -160,12 +178,15 @@ static List<Case> goodCases() {
             Merge.class,
             Json.from("fromRefName", "testBranch")
                 .addNoQuotes("keepIndividualCommits", "true")
-                .add("fromHash", HASH)),
+                .add("fromHash", HASH)
+                .addNoQuotes("isFetchAdditionalInfo", null)
+                .addNoQuotes("isDryRun", null)),
```
[the remark this answers is not in the document]
Comment (snazy) [reply model/src/test/java/org/projectnessie/model/TestModelObjectsSerialization.java:183]: That didn't work - at least not before the change with the JsonParser. Let me try again.

[the remark this answers is not in the document]
Comment (snazy) [reply model/src/test/java/org/projectnessie/model/TestModelObjectsSerialization.java:183]: Yea - still needed. They're present.

File: servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImpl.java [hunk 3]
```diff
@@ -399,7 +402,9 @@ public void transplantCommitsIntoBranch(
               commitMetaUpdate(),
               keepIndividual,
               keyMergeTypes(transplant),
-              defaultMergeType(transplant));
+              defaultMergeType(transplant),
+              dryRun != null && dryRun,
```
[the remark this answers is not in the document]
Comment (snazy) [reply servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImpl.java:406]: heh - yea

File: versioned/spi/src/test/java/org/projectnessie/versioned/TestMetricsVersionStore.java [hunk 4]
```diff
@@ -104,6 +104,12 @@ public String squash(List<String> metadata) {
           }
         };
 
+    MergeResult<Object> dummyMergeResult =
```
[the remark this answers is not in the document]
Comment (snazy) [reply versioned/spi/src/test/java/org/projectnessie/versioned/TestMetricsVersionStore.java:107]: maybe not - it's only used in this test case

**Review by dimas-b — APPROVED** [review]
